modules/mobile/app/models/mobile/v0/user_accessible_services.rb

```diff
@@ -17,6 +17,14 @@ def available
         service_auth_map.keys.sort
       end
 
+      def available_and_authorized
```

Comment:
The `service_auth_map` method below forms exactly this, then we decompose it into available and authorized, then reform it. I'd suggest we just make `service_auth_map` public (and perhaps rename it if you can think of a better name). This will require substantial unit test changes, but they shouldn't take very long. IMO all of the testing complexity should be for the `service_auth_map` method, then the `authorized` method unit tests can be a single "it forms an array of authorized services".